design/space_views.md

```diff
@@ -0,0 +1,152 @@
+# Space Views
+Status: Mostly implemented.
+
+
+## What are Space Views
+Space Views visualize a Data Blueprint, i.e. a set of entities with given properties.
+They are represented as freely arrangeable tiles in the Viewport.
+Most Space Views are interactive, allowing their data to be explored freely.
+
+
+## Properties of a Space View
+All properties are saved as part of the blueprint.
+
+Immutable:
+* Space View Class
+* root entity path
+
+Mutable:
+* name
+* positioning within layout
+* class specific properties
+* Data Blueprint
+
+## Role of the root entity path
+* root of the transform hierarchy (if any is used)
+* may govern heuristics
+* available at various stages of scene build-up (see below)
+
+## Space View Class
+Each Space View refers to an immutable Space View Class, implemented by `SpaceViewClass`.
+It defines:
+* which data it can display and how it is displayed
+* how it is interacted with
+* what properties are read from the blueprint store and how they are exposed in the ui
+
+### What Space View Classes are there?
+Space View differ only in class when they are **fundamentally different** in the way they display data.
+Naturally, this means that there are only ever very few distinct Space View classes.
+
+As of writing we have:
+* Spatial
+* Bar Chart
+* Tensor
+* Text
+* Text Box
+* Time Series
+
+Proposal for future builtin, plus some notes:
+* Spatial 2D
+  * encompasses Tensor and Bar Chart
+  * can display everything Spatial 3D can display
```

Comment:
Personally I'm leaning towards keeping charts separate. However, as Niko pointed out in the past you may want to build up your own chart from the very same primitives that you were to place in a regular 2D view.
Once you're there you might as well put 3D things in your chart: E.g. axis around a mesh?
That said, with things like that I'm struggling again with why we should make a difference between 2D and 3D in the first place. (my own initial intuition is that 2D and 3D are the same but charts are something else)

Reply:
### On merging Bar Chart with Spatial 2D
I think it might take some time to get the Archetype Queries + defaults expressive and and easy to use enough that it makes sense to merge bar chart with spatial 2D. Right now we have the state that the bar chart space view takes a single 1-D tensor and draws a bar chart with x-axis = tensor indices and y-axis = tensor values. It draws boxes with width 1, centered on integers in x, y-min = 0 and y-max = tensor value.

With the right set of primitives a user should be able to manually build a bar chart in a spatial 2D view. For example they might want a stacked bar chart. Talking about bringing in 3D into a bar chart doesn't likely make sense since there probably doesn't exist a camera projection that maps between 3D and the tensor indices axis (x).

One could imagine that we would have heuristics that generate a Data Blueprint for boxes that creates a bar chart from 1-D tensors.

### On why 2D and 3D space views shouldn't be the same
It is true that 2D and 3D space views are very similar from a rendering point of view. However, they are more different from a users point of view. 

First of all 3D data is only viewable in 2D if combined with a suitable projection (could be through perspective projection or by dropping the data of one dimension). The fact that 2D views are rendered in a 3D pipeline using some kind of pseudo depth (draw order) and an implicitly defined orthographic camera, is not top of mind to me as a user. This is something you need considerable exposure to graphics or 3D computer vision to experience as immediately obvious.

Second, my expectations around how to navigate a 2D visualization are quite different from how I expect to navigate a 3D visualization.

### The fundamental difference between different space views
In my mind, the fundamental difference between different space views lies in the kinds of axes a view has.
- Data Table (currently text views) have rows and columns with text
- Text Log have rows with logs sorted in time (not 100% sure this is fundamentally different than Data Table)
- Spatial 2D has two orthogonal axes with defined spatial relationships
- Spatial 3D has three orthogonal axes with defined spatial relationships
- Time Series has one time axis and one numeric axis
- Rich Text is a rich text document (linear in top to bottom with wraparound in horizontal)

Reply:
excellent writeup @nikolausWest , thank you! This makes perfectly sense to me and I'll take it over into the RFC document. I like the guidance of "has different axis" as a rough principal.
(This also doubles down that we need to make it easy to share SceneParts/ArchetypeProcessors or parts thereof between SpaceView classes.)